frontends/concrete-python/concrete/fhe/compilation/circuit.py

```diff
@@ -642,4 +499,55 @@ def statistics(self) -> Dict:
         """
         Get all statistics of the circuit.
         """
-        return self._property("statistics")  # pragma: no cover
+        mod_stats = self._module.statistics
+        func_stats = mod_stats.pop("functions")[self._name]
+        return mod_stats | func_stats  # pragma: no cover
+
+    @property
+    def configuration(self) -> Configuration:
+        """
+        Return the circuit configuration.
+        """
+        return self._module.configuration
+
+    @property
+    def graph(self) -> Graph:
+        """
+        Return the circuit graph.
+        """
+        return self._function.graph
+
+    @property
+    def mlir_module(self) -> MlirModule:
+        """
+        Return the circuit mlir module.
+        """
+        return self._module.mlir_module
+
+    @property
+    def compilation_context(self) -> CompilationContext:
+        """
+        Return the circuit compilation context.
+        """
+        return self._module.compilation_context
+
+    @property
+    def client(self) -> Client:
+        """
+        Return the circuit client.
+        """
+        return self._module.execution_runtime.val.client
+
+    @property
+    def server(self) -> Server:
+        """
+        Return the circuit server.
+        """
+        return self._module.execution_runtime.val.server
+
+    @property
+    def simulator(self) -> Server:
+        """
+        Return the circuit simulator.
+        """
+        return self._module.simulation_runtime.val.server
```

Comment:
self._module.simulator ? btw is that accesor is really usefull as there are no access to the simulation client?

Reply:
Good question. Probably not so useful yes. I’ll have a look in the refactoring PR if that’s ok for you !